src/averageAges.js

```diff
@@ -55,7 +69,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const hasMother = (group, person) => {
+    const mother = group.find(mthr => mthr.name === person.mother);
+
+    return mother;
+  };
+
+  const ageDiff = [...people].filter((person, idx, arr) => {
```

Comment:
filter is non-mutating method

Reply:
where have you used index? why do you need this array from filter if you have `people`?